pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -328,6 +329,13 @@ def _apply_state_vector(self, state, device_wires):
                     or broadcasted state of shape ``(batch_size, 2**len(wires))``
                 device_wires (Wires): wires that get initialized in the state
             """
+
+            if isinstance(state, self._qubit_state.__class__):
+                state_data = np.zeros(state.size, dtype=self.C_DTYPE)
+                state_data = self._asarray(state_data, dtype=self.C_DTYPE)
```

Comment:
Do we need to initialise the 0s first and then handle the conversion in the following line? Could we call `allocate_aligned_array(.., True)` directly to create an aligned and 0-init'd memory block to use?

Reply:
It's not needed, I was following LKokkos here. I updated it and removed an unnecessary ravel, in both LQubit and LKokkos.